abci/src/tracing_span.rs

```diff
@@ -38,20 +41,28 @@ macro_rules! block_span {
 /// # Examples
 ///
 /// ```
-/// # use tenderdash_proto::abci::{RequestInfo, request};
+/// # use tenderdash_proto::abci::{RequestInfo, Request, request::Value};
 /// # use tenderdash_abci::tracing_span::span;
 ///
-/// let request = request::Value::Info(RequestInfo::default());
-/// let span = span(request);
+/// let request = Request {
+///    value: Some(Value::Info(Default::default())),
+/// };
+/// let span = span(&request);
 /// ```
-pub fn span<T>(request: T) -> tracing::span::EnteredSpan
+pub fn span(request: &abci::Request) -> tracing::span::EnteredSpan
 where
-    T: Into<Value>,
 {
-    let value = request.into();
+    let value = request.value.as_ref().expect("request value is missing");
+
+    // we use md5 as we need 16-byte request id for uuid, and it doesn't have to be
+    // cryptographically secure
+    let mut md5 = lhash::Md5::new();
```

Comment:
I think it's pretty expensive, especially for the process and prepare proposal that contain state transitions. Should we do that only if a specific logging level is enabled?